### DISCUSSION TITLE
FastAPI @app.get inside NiceGUI page

### DISCUSSION THREAD

**Discussion by Nikodemski2002** [zauberzeug/nicegui#507] [Q&A] [answered]
Hi, first of all, I want to thank you for this framework and the work of all its contributors, it's really wonderful.
I have a question regarding the use of @app.get from FastAPI inside the NiceGUI page. Namely, I have a home_page.py file that contains the page content and I want to render this file content with or without a notification based on the ?success=true or 
?success=false statement. I think it would be more understandable on a code example:

So this is my main.py file:
```py
from nicegui import app, ui
import home_page

@ui.page('/home')
def home() -> None:
    home_page.content()

ui.button('Login', on_click= lambda: ui.open('/home?success=true')).props('icon=login')
ui.run(port=8585)
```
And this is my home_page.py file:
```py
from nicegui import ui

def content() -> None:
    ui.markdown('###Home Page')
```
The thing I try to achieve is when the URL localhost:8585/home?success=true is opened the page will show together with a notification: 'Successfully logged in', in the opposite when the URL localhost:8585/home?success=false is opened then only page content will show without any notification. I tried changing the home_page.py code to the version shown below:
```py
from nicegui import app, ui

def content() -> None:
    @app.get('/home')
    def check(success: bool):
        if success:
            ui.notify('Successfully logged in')
            ui.markdown('###Home Page')
        else:
            ui.markdown('###Home Page')
```
but it didn't work. 

So **is it possible to call a function inside the page .py file, based on query parameters parsed together with a page URL**, so it will affect the way the GUI renders? If something I've mentioned here isn't clear or understandable feel free to ask me.
Kind regards,
Nikodem

**Comment by falkoschindler** [comment, marked as answer]
Yes, that's possible. But you need to handle the query parameter in the page function and pass it to `content`. Here is an example:

```py
def content(success: bool) -> None:
    ui.markdown('### Home Page')
    if success:
        ui.label('Successfully logged in')

@ui.page('/home')
def home(success: bool) -> None:
    content(success)
```

Note that this example uses a label instead of `ui.notify`. To show a notification immediately after opening a page, we need to wait for the client connection. Therefore we add a `client` parameter and await its `connected` method. Then we can call `content` and the notification is successfully sent to the client:

```py
from nicegui import Client

def content(success: bool) -> None:
    ui.markdown('### Home Page')
    if success:
        ui.notify('Successfully logged in')

@ui.page('/home')
async def home(success: bool, client: Client) -> None:
    await client.connected()
    content(success)
```

**Reply by Nikodemski2002** [reply]
Thank You so much, I think I've tried something simillar but in a totally wrong way 😅.

Have a great day!
Nikodem
